.ci/Jenkinsfile

```diff
@@ -282,31 +288,89 @@ pipeline {
+            skipDefaultCheckout()
           }
-          steps {
+          steps{
             deleteDir()
-            dir("${OPBEANS_REPO}"){
-              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
-                  url: "git@github.com:elastic/${OPBEANS_REPO}.git"
-              // It's required to transform the tag value to the artifact version
-              sh script: ".ci/bump-version.sh ${env.BRANCH_NAME.replaceAll('^v', '')}", label: 'Bump version'
-              // The opbeans pipeline will trigger a release for the master branch
-              gitPush()
-              // The opbeans pipeline will trigger a release for the release tag
-              gitCreateTag(tag: "${env.BRANCH_NAME}")
+            unstash 'source'
+            // Bail out if we don't have the right parameters
+            isValidReleaseVer("${params.release_ver}")
+            sh((label: 'Prepare release toolchain', script: '.ci/scripts/prepare-release-containers.sh')
+            // Generate a preview of the CHANGELOG file for visual inspection
+            script{
+              def credentialsId = params.get('credentialsId', '2a9602aa-ab9f-4e52-baf3-b71ca88469c7')
+              withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
+                def preview = sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview true", returnStdout: true)
+              }
+              input(message: 'Do you accept these proposed entries for the CHANGELOG.asciidoc file?', ok: 'Yes, proceed with writing them.',
+                    parameters: [text(description: 'IMPORTANT: Proposed changes CANNOT be edited here. Abort the build if not acceptable.',
+                    defaultValue: "${preview}", name: 'preview')])
+              withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
+                // Re-run without preview flag and rewrite the file
+                sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview false > ./CHANGELOG.asciidoc", returnStdout: true)
+              }
+            }
+            // Update the EOL table in the upgrading.asciidoc file
+            sh(label: 'Update EOL table', script: ".ci/scripts/generate_eol.sh --version ${params.release_ver} > ./docs/upgrading.asciidoc")
+            sh(label: 'Update package.json', script: ".ci/scripts/update_package_version.sh ${params.release_ver}")
+                withCredentials([usernamePassword(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken',
+                                      passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+                                        sh(label: 'Tag and push', script: ".ci/scripts/tag_release.sh --version ${params.release_ver}")
+                                      }
+            
+            dir("${BASE_DIR}") {
```

Comment:
this particular `dir` should be added to the very beginning [L311](https://github.com/elastic/apm-agent-nodejs/pull/1714/files#diff-8db37e8fcea0f1a8f2f39667e94ebcc4R311)